roles/azimuth_capi_operator/tasks/main.yml

```diff
@@ -72,6 +72,26 @@
   loop_control:
     label: "{{ item.metadata.name }}"
 
+- name: Remove existing access control annotations on cluster templates
+  command: >-
+    kubectl annotate --all --overwrite
+    clustertemplates.azimuth.stackhpc.com
+    {{ item }}-
+  loop:
+    - "acl.azimuth.stackhpc.com/allow-list"
+    - "acl.azimuth.stackhpc.com/deny-list"
+    - "acl.azimuth.stackhpc.com/allow-regex"
+    - "acl.azimuth.stackhpc.com/deny-regex"
+
+- name: Ensure cluster template tenancy access controls are applied
+  command: >-
+    kubectl annotate --all --overwrite
+    clustertemplates.azimuth.stackhpc.com
+    {{ item.key }}={{ item.value }}
+  loop: "{{ azimuth_capi_operator_cluster_templates_tenancy_access_controls | dict2items }}"
```

Comment:
Do these tasks leave a period proportional to the number of templates available where all annotations may be removed from a clustertemplate? If so, slim chance of causing issues, but clustertemplates may briefly show as available to users...

Not completely sure of the loop conditions required to prevent this - but maybe we need to go clustertemplate-wise, so remove and add acls in a single loop iteration. Even better if we can loop clustertemplate-and-acl-wise, but I'm not sure of the practicality of that.

Reply:
This is a good point, as you say it's a pretty small window where it could happen, but I guess erroring out on the second task would do the same but for much longer... I'll have a think about this one.